src/app/component/ProductGallery/ProductGallery.component.js

```diff
@@ -64,14 +75,33 @@ export default class ProductGallery extends PureComponent {
         />
     );
 
+    handleZoomStart = () => {
+        const { isZoomEnabled } = this.state;
+        if (isZoomEnabled) return;
+
+        document.body.classList.add('overscrollPrevented');
+        this.setState({ isZoomEnabled: true });
+    };
+
+    disableZoom = () => {
+        document.body.classList.remove('overscrollPrevented');
+        this.setState({ isZoomEnabled: false });
+    };
+
+    handleZoomChange = (args) => {
+        if (args.scale > 1) {
+            this.handleZoomStart();
+        }
+    };
```

Comment:
I think we must create a container for ProductGallery and move all non render functions there. Along with a state. Then, the component props wont update too often.